tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/AwsSqsCommon.cs

```diff
@@ -56,5 +59,28 @@ public static Scope CreateScope(Tracer tracer, string operation, out AwsSqsTags
             // or we couldn't populate it completely (some tags is better than no tags)
             return scope;
         }
+
+        public static string GetQueueName(string queueUrl)
+        {
+            if (queueUrl is null)
+            {
+                return queueUrl;
+            }
+
+            return queueUrl.Split('/').Last();
+        }
+
+        internal static string GetOperationName(Tracer tracer, string spanKind) => tracer.CurrentTraceSettings.Schema.Version switch
+        {
+            SchemaVersion.V0 => SqsRequestOperationName,
+            _ => GetOperationNameHelper(spanKind)
+        };
+
+        internal static string GetOperationNameHelper(string spanKind) => spanKind switch
+        {
+            SpanKinds.Consumer => $"{SnsOperationName}.process",
+            SpanKinds.Producer => $"{SnsOperationName}.send",
+            _ => $"{SnsOperationName}.request"
```

Comment:
SHould we throw here? Or at least do you throw in ITests if the kind is different from producer, consumer or client?

Reply:
Exactly what Pablo said. I have commands which are neither producer, nor consumer. So I decided to stick to what we had before.

I also asked in the APM Naming Working Group and they suggested to stick with `aws.sqs.request`.